packages/wdio-config/src/lib/ConfigParser.ts

```diff
@@ -247,7 +257,7 @@ export default class ConfigParser {
         if (filesToFilter.size === 0) {
             throw new Error(`spec file(s) ${cliArgFileList.join(', ')} not found`)
         }
-        return filesToFilter
+        return [...filesToFilter]
```

Comment:
Any reason why we transform it back to an array here? I have no strong opinion whether to use a Set or array but if we transform here we shouldn't do it again at places where `setFilePathToFilterOptions` is used.

Reply:
I do not have a strong opinion on this either. The reasons I chose to convert it into strings are:
1. The JSDoc comment above states the return type to be `String[]`.
2. It semantically feels a bit weird to me that a function that takes an array of filenames to filter, would return a `Set` type. The uniqueness property is an implementation-specific thing, and the function return type shouldn't be implementation-specific, I think. That's why I changed it to `string[]`. I'm okay to change it back if you insist.

Btw, you're right about not to transform again where `setFilePathToFilterOptions` is used. I will make that change.

Reply:
If we return with an array let's define `const filesToFilter: string[] = []` then

Reply:
Oh I thought we want to avoid duplicates, that's why we are using Set internally in that function

Reply:
I'm getting this TS error:
```
This expression is not callable.
  Each member of the union type '{ <S extends (config: Config, capabilities: DesiredCapabilities[]) => void>(predicate: (value: (config: Config, capabilities: DesiredCapabilities[]) => void, index: number, array: ((config: Config, capabilities: DesiredCapabilities[]) => void)[]) => value is S, thisArg?: any): S[]; (predicate: (value: (config: Confi...' has signatures, but none of those signatures are compatible with each other.ts(2349)
```

Reply:
ah makes sense, nevermind then